### ISSUE TITLE
Linting fails when running test specs on older simulator

### ISSUE THREAD

**Issue by armcknight** [CocoaPods/CocoaPods#9211]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

pod spec lint PippinLibrary.podspec

## What did you expect to happen?

Linting should have succeeded.

## What happened instead?

Linting failed with an error response from xcodebuild.

After digging in more using `--no-clean` and messing about with the generated xcworkspace, I was able to discover that the test bundle always failed to install to a simulator running an earlier version of iOS. I have simulators with iOS 10.3.1 installed (in addition to the ones running iOS 13.1) and they all fail the same way:
```
Testing failed:
	PippinLibrary-Unit-Tests:
		xctest (3115) encountered an error (Failed to load the test bundle. (Underlying error: The bundle “PippinLibrary-Unit-Tests” couldn’t be loaded because it is damaged or missing necessary resources. The bundle is damaged or missing necessary resources. dlopen_preflight(/Users/andrew/Library/Developer/Xcode/DerivedData/App-hkgnrgwjwqsxzdbyrdupojngaynu/Build/Products/Debug-iphonesimulator/PippinLibrary-Unit-Tests.xctest/PippinLibrary-Unit-Tests): Library not loaded: @rpath/libswiftUIKit.dylib
  Referenced from: /Users/andrew/Library/Developer/Xcode/DerivedData/App-hkgnrgwjwqsxzdbyrdupojngaynu/Build/Products/Debug-iphonesimulator/PippinLibrary/PippinLibrary.framework/PippinLibrary
  Reason: no suitable image found.  Did find:
	/usr/lib/swift/libswiftUIKit.dylib: mach-o, but not built for iOS simulator
	/usr/lib/swift/libswiftUIKit.dylib: mach-o, but not built for iOS simulator))
```

Now I don't think this is necessarily a CocoaPods issue. But I am curious how you go about choosing which simulator to use to run tests in a pod's `test_spec`? It seems like it takes the first in the list supplied by `simctl`. Is there any way to specify/override the `-destination` flag supplied to `xcodebuild`?

### Stack

```
   CocoaPods : 1.8.1
        Ruby : ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-darwin18]
    RubyGems : 2.7.6.2
        Host : Mac OS X 10.14.6 (18G103)
       Xcode : 11.1 (11A1027)
         Git : git version 2.18.0
Ruby lib dir : /Users/andrew/.rbenv/versions/2.5.5/lib
Repositories : <redacted>
               master - git - https://github.com/CocoaPods/Specs.git @ 0f634a4db9d4cb713ea8f671659e316f0cd0ea59
               trunk - CDN - https://cdn.cocoapods.org/
               tworingsoft - git - https://gitlab.com/amcknight/tworingsoft-podspecs.git @ f34ebfc6e396710f67f357211d7fef56e36d80bf
               <redacted>
```

### Installation Source

```
Executable Path: /Users/andrew/Work/TwoRingSoft/Frameworks/Pippin/vendor/bundle/ruby/2.5.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

The project can be found at https://github.com/tworingsoft/pippin.

**Comment by dnkoutso** [comment]
The warning message appears to be the same issue as https://github.com/CocoaPods/CocoaPods/issues/9165.

Going to preemptively close as duplicate for now but we can re-open it. From https://github.com/CocoaPods/CocoaPods/issues/9165 it seems that iOS 13 works and iOS 11 doesn't. Noone has confirmed why/how this happens but might be an Apple/Xcode issue.

**Comment by dnkoutso** [comment]
Thank you by the way for the sample app and the detailed report!

**Comment by armcknight** [comment]
Thanks for the info... I did try searching beforehand, promise, just didn't see those!

For what it's worth, I just set up a similar new project to the one created by `spec lint` and looked at the diff between the two. One thing that jumped out at me was `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES`. It was `No` in the pod project, and `Yes` in my test project. After recreating my iOS 10.3.1 simulator, I changed it to `Yes` in the pod project and the tests were able to run on it.

Maybe there's something going on, now that Swift is supposed to ship at the OS level due to ABI stability, when things get run on the simulator. I looked in the test bundle created by the pod project and there are still swift dylibs in it. But, I figured that wouldn't happen and that it should load from the macOS versions on my machine, which it appears to want to do in my error message: it's talking about a dylib in `/usr/lib/`. ¯\_(ツ)_/¯ This is guesswork on my part.

I added this to my podspec:
```ruby
s.xcconfig = { 'ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES' => 'Yes' }
```
and confirmed that the `spec lint` now passes, even using my old simulator to run the test specs.
